### PR TITLE
Add debugging for auth, fix heartbeat initialization and bundling

### PR DESCRIPTION
**Pull request by jrw972**

### CODE REVIEW

File: dds/DCPS/RTPS/Spdp.cpp [hunk 1]
```diff
@@ -1174,12 +1202,32 @@ Spdp::handle_handshake_message(const DDS::Security::ParticipantStatelessMessage&
...
+               DCPS::LogGuid(src_participant).c_str(),
+               DCPS::LogGuid(msg.destination_participant_guid).c_str(),
+               DCPS::LogGuid(guid_).c_str()));
+  }
+
   // If this message wasn't intended for us, ignore handshake message
-  if (msg.destination_participant_guid != guid_ || !msg.message_data.length()) {
+  if (msg.destination_participant_guid != guid_) {
+    if (DCPS::security_debug.auth_debug) {
+      ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) {auth_debug} DEBUG: Spdp::handle_handshake_message() - ")
```
Comment (jwillemsen) [inline dds/DCPS/RTPS/Spdp.cpp:1218]: Log the guids?

File: dds/DCPS/RTPS/Spdp.cpp [hunk 2]
```diff
@@ -915,6 +939,10 @@ Spdp::handle_auth_request(const DDS::Security::ParticipantStatelessMessage& msg)
 
   if (iter != participants_.end()) {
     if (msg.message_identity.sequence_number <= iter->second.auth_req_sequence_number_) {
+      if (DCPS::security_debug.auth_debug) {
+        ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) {auth_debug} DEBUG: Spdp::handle_auth_request() - ")
+                   ACE_TEXT("Dropped due to old sequence number\n")));
```
Comment (jwillemsen) [inline dds/DCPS/RTPS/Spdp.cpp:944]: Log the sequence numbers?

File: dds/DCPS/RTPS/Spdp.cpp [hunk 3]
```diff
@@ -895,18 +895,42 @@ Spdp::match_unauthenticated(const DCPS::RepoId& guid, DiscoveredParticipantIter&
...
+    return;
+  }
 
   ACE_GUARD(ACE_Thread_Mutex, g, lock_);
 
   if (sedp_->ignoring(guid)) {
     // Ignore, this is our domain participant or one that the user has
     // asked us to ignore.
+    if (DCPS::security_debug.auth_debug) {
+      ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) {auth_debug} DEBUG: Spdp::handle_auth_request() - ")
```
Comment (jwillemsen) [inline dds/DCPS/RTPS/Spdp.cpp:931]: Log the guid?

File: dds/DCPS/RTPS/Spdp.cpp (hunk 1 above)
Comment (jrw972) [reply dds/DCPS/RTPS/Spdp.cpp:1218]: Guids will be logged on entry into the function.

File: dds/DCPS/RTPS/Spdp.cpp (hunk 3 above)
Comment (jrw972) [reply dds/DCPS/RTPS/Spdp.cpp:931]: Guids are logged upon entry into the function.

File: dds/DCPS/RTPS/Spdp.cpp (hunk 2 above)
Comment (jrw972) [reply dds/DCPS/RTPS/Spdp.cpp:944]: How would a user meaningful use the sequence numbers?

**Review by iguessthislldo — APPROVED** [review]

File: dds/DCPS/RTPS/Spdp.cpp (hunk 2 above)
Comment (jwillemsen) [reply dds/DCPS/RTPS/Spdp.cpp:944]: Would help to understand why the log message is there in the log

**Review by mitza-oci — APPROVED** [review]

**Pull request merged**